City-of-Helsinki/drupal-helfi-etusivu
# UFH-5914 Filters

# [UHF-5914 News archive filters](https://helsinkisolutionoffice.atlassian.net/browse/UHF-5914)
Feature for adding filters to news archive app.

## What was done
* Added upgraded version of news archive app
  * Added capability to filter results via dropdowns
* Read more here: https://github.com/City-of-Helsinki/helfi-etusivu-news-search/pull/4

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout/UHF-5914-filters`
  * `make fresh`
* Make sure your elastic is up and running and has news indexed to it:
  * Run `docker-compose ps` and seet that the elastic container is up
  * Check the index management page: https://helfi-etusivu.docker.so/en/admin/config/search/search-api/index/news
    * Run `rebuild tracking information`
    * Run `Index now`
* `make shell` into container
  * Run `drush locale:check` && `drush locale:update` to get translations
* Run `make drush-cr`

## How to test
* Navigate to the news page: https://helfi-etusivu.docker.so/fi/uutiset
* Search functionality things to test for:
  * Filters components should work
  * Filters and results should consider the language context
  * Query should be executed only when pressing the submit button OR the 'clear all' button
  * No results should render 'no results' text
  * Changing the filter values should be represented in the url query string

**Things to consider when testing**:
* Fallback listing and react app are being displayed at the same time for now.. to be fixed in an upcoming ticket. Meaning there are _two same looking news lists_. This PR is for testing the top-one (with filters).
* Making "Most read news" -block work with react-app needs work (upcoming ticket) so the search app is 100% width for now.
* Filter pills are not implemented yet.
* Some translations are missing (pending PO).


## Designers review



## Other PRs


*https://github.com/City-of-Helsinki/helfi-etusivu-news-search/pull/4
